src/ncp/ncp_base_mtd.cpp

```diff
@@ -2975,6 +2977,24 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_CNTR_ALL_IP_COUNTERS>
     return OT_ERROR_NONE;
 }
 
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_STREAM_CLI>(void)
+{
+    otError     error = OT_ERROR_NONE;
+    char        buffer[OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH];
+    const char *string = nullptr;
+    size_t      length = 0;
+
+    SuccessOrExit(error = mDecoder.ReadUtf8(string));
+    length = strlen(string);
+    VerifyOrExit(length < OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH, error = OT_ERROR_INVALID_COMMAND);
+    memcpy(buffer, string, length + 1);
```

Comment:
Could we use `const_cast` to convert the `string` to `char *` and then pass it to `otCliInputLine ()` to avoid temporary allocating memory and memory copy?